### PR TITLE
Chi-square test

### LINKED ISSUE #8: Chi-square test

### PR DESCRIPTION
**Pull request by sergey-jr**
https://github.com/Innopolis-University-Adv-Statistics/Interactive-Statistics-Notebooks/issues/8

### CODE REVIEW

**Review by apoyezzhayev — COMMENTED** [review]
The notebook uses real-world data and have visualizations. However, there are few issues to pay attention on.  
1) Not convenient to use drop-down for range arguments in `@interact` better to use sliders or text fields
2) No formulas at all. It will be great if at least basic tests be backed with latex math formulas.
3) The information about other chi-squared tests is not present:
  - Chi-squared test for variance in a population distributed normally - important
  - Chi-square test for normality - special case of goodness of fit
  - Binomial test - special case of goodness of fit
  - Reference to Fisher's test 
  - At least links to other tests from wiki using the chi-squared test 
4) There are only two examples and no tasks at all.
5) There are some uncertainties in definitions which should be fixed. 
6) For every hypothesis testing it's better to stick with standard description pipeline like in lectures and labs:
  - Check if assumptions hold
  - Define null and alternative hypothesis
  -  Compute test statistics
  - Compare with table value
  - How to make a decision

File: Chi_square_test.ipynb [hunk 1]
```diff
@@ -0,0 +1,1190 @@
...
+        "colab_type": "text"
+      },
+      "source": [
+        "## Intro\n",
+        "\n",
+        "The term \"chi-squared test,\" also written as χ<sup>2</sup> test, refers to certain types of statistical hypothesis tests that are valid to perform when the test statistic is chi-squared distributed under the null hypothesis. Often, however, the term is used to refer to Pearson's chi-squared test and variants thereof. Pearson's chi-squared test is used to determine whether there is a statistically significant difference (i.e., a magnitude of difference that is unlikely to be due to chance alone) between the expected frequencies and the observed frequencies in one or more categories of a so-called contingency table.\n",
+        "\n",
+        "There are two types of chi-square tests. Both use the chi-square statistic and distribution for different purposes:\n",
+        "\n",
+        "- A chi-square goodness of fit test determines if a sample data matches a population.\n",
```
Comment (apoyezzhayev) [inline Chi_square_test.ipynb]: What does this "sample matches population" mean? Some pre-defined distribution?

Comment (apoyezzhayev) [reply Chi_square_test.ipynb]: It's better to add the information that this test has another name - Pearson's chi-squared test

Comment (apoyezzhayev) [reply Chi_square_test.ipynb]: Why there is no any information about assumptions, degrees of freedom and formulas.

File: Chi_square_test.ipynb [hunk 2]
```diff
@@ -0,0 +1,1190 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "q5XnPEsm0phq",
+        "colab_type": "text"
+      },
+      "source": [
+        "# METADATA\n",
```
Comment (apoyezzhayev) [inline Chi_square_test.ipynb]: Why do we need it at all? Seems like a copy-paste from the template

**Review by apoyezzhayev — CHANGES_REQUESTED** [review]

**Comment by sergey-jr** [comment]
@apoyezzhayev 
- [V] Not convenient to use drop-down for range arguments in @interact better to use sliders or text fields
- [added] No formulas at all. It will be great if at least basic tests be backed with latex math formulas. 
- [added] The information about other chi-squared tests is not present

**Comment by Lehsuby** [comment]
Pros:
- Structure good
- Link to colab
- Nice examples
- Good visualization part

Cons:
- There are **no tasks, self-checking questions**
- The **dependence** of the test on the **number of examples** is not shown
- **Confirmation of a hypothesis** differs from the algorithm that is used in labs

Advices:
- correct spelling mistakes
- add more examples, exercices, self-checking questions
- add examples, which can confirm all statements from the conclusion
- add correct definition of hypothesis

File: Chi_square_test.ipynb [hunk 3]
```diff
@@ -0,0 +1,1190 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "t3CJF7GLQ6bK",
+        "colab_type": "text"
+      },
+      "source": [
+        "Example 2. \n",
+        "\n",
+        "We have some sets of data, each of them consisits from ~250 elements. We want to check theitr dictribution. The distributions we wil chick on ot normal, uniform and longnorm."
```
Comment (Lehsuby) [inline Chi_square_test.ipynb]: dictribution, chick

**Review by Lehsuby — CHANGES_REQUESTED** [review]
Pros:
- Structure good
- Link to colab
- Nice examples
- Good visualization part

Cons:
- There are **no tasks, self-checking questions**
- The **dependence** of the test on the **number of examples** is not shown
- **Confirmation of a hypothesis** differs from the algorithm that is used in labs

Advices:
- correct spelling mistakes
- add more examples, exercises, self-checking questions
- add examples, which can confirm all statements from the conclusion
- add correct definition of hypothesis

**Comment by sergey-jr** [comment]
@Lehsuby , @apoyezzhayev , @jacked97 please check

**Review by Lehsuby — CHANGES_REQUESTED** [review]
Cons:

**There are no tasks, self-checking questions**
**Confirmation of a hypothesis** differs from the algorithm that is used in labs

Advices:

**correct spelling mistakes** (comment above)
add exercises, self-checking questions
add correct definition of hypothesis for all exercises

**Review by jacked97 — APPROVED** [review]

**Comment by sergey-jr** [comment]
@Lehsuby , @apoyezzhayev please check

**Review by Lehsuby — CHANGES_REQUESTED** [review]
Please, add self-checking questions after examples and exercises

**Review by Lehsuby — APPROVED** [review]
Good, you can add @iteract module for exercises, if you want

**Comment by sergey-jr** [comment]
@jacked97, @apoyezzhayev  please check if all is good.

**Review by apoyezzhayev — APPROVED** [review]

**Review by jacked97 — APPROVED** [review]

**Pull request merged**